libraries/chain/include/eosio/chain/types.hpp

```diff
@@ -410,20 +409,78 @@ namespace eosio::chain {
 
 }  // eosio::chain
 
+
+namespace fc::raw {
+   // pack/unpack chainbase::shared_cow_string
+   // ----------------------------------------
+   template<typename Stream>
+   inline void pack(Stream& s, const chainbase::shared_string& v) {		
+      FC_ASSERT(v.size() <= MAX_SIZE_OF_BYTE_ARRAYS);		
+      fc::raw::pack(s, fc::unsigned_int((uint32_t)v.size()));		
+      if (v.size())
+         s.write(v.data(), v.size());		
+   }		
+
+   template<typename Stream>
+   inline void unpack(Stream& s, chainbase::shared_string& v) {
+      fc::unsigned_int sz;
+      fc::raw::unpack(s, sz);
+      if (sz) {
+         v.resize_and_fill(sz, [&](char* buf, std::size_t sz){
+            s.read(buf, sz);            
+         });
+      }
+   }
+   
+   // pack/unpack chainbase::shared_cow_vector
+   // ----------------------------------------
+   template<typename Stream, typename T>
+   inline void pack(Stream& s, const chainbase::shared_cow_vector<T>& v)  {
+      FC_ASSERT(v.size() <= MAX_SIZE_OF_BYTE_ARRAYS);
```

Comment:
The `MAX_SIZE_OF_BYTE_ARRAYS` is for bytes where used elsewhere. This should be specialized for `char` type if we want to enforce this.

Reply:
Is [this](https://github.com/AntelopeIO/leap/pull/1851/commits/4b011f3aa088fcd8e4e48bfa704c3a38f8b2a0e4) what you meant?

Reply:
No, I mean you need this impl: https://github.com/AntelopeIO/leap/blob/dd4217f111619f758a13e5d06a95a84f179058f8/libraries/libfc/include/fc/container/flat.hpp#L13-L13

And the specializations for `char`

Reply:
Is this really needed? As far as I can tell, these are just optimizations, and we don't serialize any shared_vector of fundamental types.

Reply:
But probably I should use `MAX_NUM_ARRAY_ELEMENTS` instead of `MAX_SIZE_OF_BYTE_ARRAYS` for `shared_vector`.

Reply:
I have updated the serialization... please let me know what you think.